api/users.rb

```diff
@@ -150,7 +150,7 @@
     raw_query: raw_query
   )
 
-  if sort_key == 'user_activity'
+  if raw_query
```

Comment:
@xitij2000 As I understand it, the purpose of extracting the sorting mechanism for `user_activity` filter is to reinstate the old sorting behaviour where the threads are sorted by latest activity of the given user instead of latest activity from any user. Is this understanding of mine correct ?

From my testing, I noticed the following behaviour:
1. Threads are sorted by latest activity from any user, given that the passed user has participated in the comment threads at some point. This is because, when any comment is posted by any user, the `updated_at` value of that particular `comment thread` always seems to be updated. So, when we sort by ` active_contents = active_contents.order_by(updated_at: :desc)` the comment threads comeup first no matter which user had done the last activity.

3. If the given user has not participated in a comment thread, then that thread is not included in the response thread list.

Reply:
For 1. That is expected, since per-user per-thread activity isn't stored exactly. This is the same as the old behaviour. 

For 2. I'm not clear what you mean. This is a user endpoint, so only threads a user has participated in should be returned.

Reply:
@xitij2000 

My main concern here was point 1. If that is the expected behaviour then I am all good.

For point 2, it totally makes sense, I was just double checking with you that it is infact the expected behaviour.